From d5a06839b51ed3a25e3143868af12fef08c7cf10 Mon Sep 17 00:00:00 2001
From: Yanling Song <songyl@ramaxel.com>
Date: Wed, 23 Feb 2022 16:53:27 +0800
Subject: [PATCH] net/spnic: Fix ethtool loopback command failure
Patch-mainline: NO, OTHERS
References: OLK-5.10
Modified-by-SEL: No


Ramaxel inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4TPKM
CVE: NA

-----------------------------------------------

Fix ethtool loopback command failure

Signed-off-by: Yanling Song <songyl@ramaxel.com>
Reviewed-by: wenliang <wenliang@ramaxel.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/ramaxel/spnic/spnic_mag_cfg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ramaxel/spnic/spnic_mag_cfg.c b/drivers/net/ethernet/ramaxel/spnic/spnic_mag_cfg.c
index 2f73f4e6cbfa..d0448d1a6bd3 100644
--- a/drivers/net/ethernet/ramaxel/spnic/spnic_mag_cfg.c
+++ b/drivers/net/ethernet/ramaxel/spnic/spnic_mag_cfg.c
@@ -178,7 +178,7 @@ int spnic_set_loopback_mode(void *hwdev, u8 mode, u8 enable)
 		return -EINVAL;
 	}
 
-	return spnic_cfg_loopback_mode(nic_cfg, MGMT_MSG_CMD_OP_GET, &mode, &enable);
+	return spnic_cfg_loopback_mode(nic_cfg, MGMT_MSG_CMD_OP_SET, &mode, &enable);
 }
 
 int spnic_set_led_status(void *hwdev, enum mag_led_type type, enum mag_led_mode mode)
-- 
2.26.2

